### PR TITLE
copr: vectorize dayofyear

### PR DESCRIPTION
**Pull request by AerysNan**
Signed-off-by: AerysNan <aerysnan@gmail.com>

PCP #5751

###  What have you changed?

Vectorize dayofyear.

###  What is the type of the changes?

Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

**Comment by AerysNan** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/62

**Comment by AerysNan** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/62

**Comment by iosmanthus** [comment]
@AerysNan Please handle the conflicts

**Comment by AerysNan** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/62

**Comment by AerysNan** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/62

**Comment by AerysNan** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/62

File: components/tidb_query/src/rpn_expr/impl_time.rs [hunk 1]
```diff
@@ -52,6 +53,22 @@ pub fn week_day(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<In
...
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn day_of_year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    if t.is_none() {
+        return Ok(None);
+    }
+    let t = t.as_ref().unwrap();
+    if t.invalid_zero() {
+        return ctx
+            .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", t)))
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_time.rs]: `Error::incorrect_datetime_value(t)` is enough.

Comment (AerysNan) [reply components/tidb_query/src/rpn_expr/impl_time.rs]: updated

**Pull request merged**